onmt/model_builder.py

```diff
@@ -150,16 +150,21 @@ def load_test_model(opt, model_path=None):
     # Avoid functionality on inference
     model_opt.update_vocab = False
 
-    model = build_base_model(model_opt, vocabs, use_gpu(opt), checkpoint,
-                             opt.gpu)
+    model = build_base_model(model_opt, vocabs, checkpoint)
+
     if opt.fp32:
         model.float()
+    elif opt.fp16:
```

Comment:
half is already applied l396, I don't understand why fp32 and fp16 are needed.
fp16, fp32 and int8 might better be a choice to make for model precision rather than true/false